packages/opds-web-client/src/reducers/history.ts

```diff
@@ -28,8 +29,11 @@ function newRootIsNotOldRoot(
   newCollection: CollectionData,
   oldCollection: CollectionData | undefined | null
 ): boolean {
-  return (
-    newCollection?.catalogRootLink?.url !== oldCollection?.catalogRootLink?.url
+  return !!(
+    oldCollection &&
+    newCollection.catalogRootLink &&
+    oldCollection.catalogRootLink &&
+    newCollection.catalogRootLink.url !== oldCollection.catalogRootLink.url
```

Comment:
Why didn't the previous implementation work?

Reply:
It's a slight thing but I found that it actually changed the semantics in this case. It's because the version with `&&` actually checks both that they are defined, and that they are not equal, whereas mine only checks that they are not equal... so if they are both undefined then they'll be equal, which isn't what we want in this case (It caused a test failure)